pkg/distributor/validate.go

```diff
@@ -232,6 +232,12 @@ func validateSampleHistogram(m *sampleValidationMetrics, now model.Time, cfg sam
 				m.maxNativeHistogramBuckets.WithLabelValues(userID, group).Inc()
 				return fmt.Errorf(maxNativeHistogramBucketsMsgFormat, s.Timestamp, mimirpb.FromLabelAdaptersToString(ls), bucketCount, bucketLimit)
 			}
+
```

Comment:
Should go above line 223, need the validation regardless if there's a bucket limit set.

Reply:
I have applied the requested change.